src/lib/realmdb/__tests__/UserProfileStorage.js

```diff
@@ -0,0 +1,455 @@
+import 'fake-indexeddb/auto'
+import fromEntries from 'object.fromentries'
+import { UserProfileStorage } from '../../userStorage/UserProfileStorage'
+import { default as goodWallet } from '../../wallet/GoodWallet'
+import getDB from '../../realmdb/RealmDB'
+import AsyncStorage from '../../utils/asyncStorage'
+
+fromEntries.shim()
+
+jest.setTimeout(30000)
+
+const profile = {
+  // avatar: '',
+  email: 'julian@gooddollar.org',
+  fullName: 'Julian Kobryński',
+  mnemonic: 'duty disorder rocket velvet later fabric scheme paddle remove phone target medal',
+  username: 'juliankobrynski',
+  mobile: '+48507471353',
+  walletAddress: '0x740E22161DEEAa60b8b0b5cDAAA091534Ff21649',
+
+  // smallAvatar: '',
+}
+
+const profileKeyValue = {
+  // avatar: '',
+  email: 'julian@gooddollar.org',
+  fullName: 'Julian Kobryński',
+  mnemonic: 'duty disorder rocket velvet later fabric scheme paddle remove phone target medal',
+  username: 'juliankobrynski',
+  mobile: '+48507471353',
+  walletAddress: '0x740E22161DEEAa60b8b0b5cDAAA091534Ff21649',
+
+  // smallAvatar: '',
+}
+
+describe('UserProfileStorage', () => {
+  let userProfileStorage
+
+  beforeAll(async () => {
+    await AsyncStorage.setItem(
+      'GD_jwt',
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJsb2dnZWRJbkFzIjoiMHg1YjliNDlmZjM1ZmE4OWZkMWZiOWNmNGJmNTNkNmI1MDA5ZmVjNjgxIiwiZ2RBZGRyZXNzIjoiMHg3NDBlMjIxNjFkZWVhYTYwYjhiMGI1Y2RhYWEwOTE1MzRmZjIxNjQ5IiwicHJvZmlsZVB1YmxpY2tleSI6IjlZdFNlSXdELVN3Z080UVIxaHBobGt4dFhleUdESjFIX01PQ3pncWcwWEkuTDN3RTJZUkpOT3c0cUo1UFVST0lRNTk3OVR3RFlCcmFmZGUwTlFkXzFSUSIsImV4cCI6MTYyOTA5Njg3MSwiYXVkIjoicmVhbG1kYl93YWxsZXRfZGV2ZWxvcG1lbnQiLCJzdWIiOiIweDViOWI0OWZmMzVmYTg5ZmQxZmI5Y2Y0YmY1M2Q2YjUwMDlmZWM2ODEiLCJpYXQiOjE2Mjg0OTIwNzF9.foR0GWU-63uJ4JIx1wBKgHs8FYVUNJvkirTWj-VCWGU',
+    )
+    const db = getDB()
+    await goodWallet.ready
+    userProfileStorage = new UserProfileStorage(goodWallet, db)
+    const seed = goodWallet.wallet.eth.accounts.wallet[goodWallet.getAccountForType('gundb')].privateKey.slice(2)
+    await db.init(seed, goodWallet.getAccountForType('gundb')) //only once user is registered he has access to realmdb via signed jwt
+  })
+
+  beforeEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  it('should not save profile without email to the db', async () => {
```

Comment:
why email/mobile/userrname are mandatory fields?
@julianpolcode @lukasz-kilaszewski 
i dont think we have any mandatory fields, we just validate they are in correct format

Reply:
This is some validation that stay with this method from gundb, and it only run if we want, to create new profile

Reply:
@lukasz-kilaszewski examine the old code carefully i do not believe we have strict requirement for all profile fields
because a user can signup with just email and no mobile

Reply:
change for throw object of field->error pair to an exception is optional
we could skip it

Reply:
@sirpy you are right, i will change it to only check for email and walletAddress